src/jbuf/jbuf.c

```diff
@@ -601,35 +611,49 @@ int jbuf_stats(const struct jbuf *jb, struct jbuf_stat *jstat)
 int jbuf_debug(struct re_printf *pf, const struct jbuf *jb)
 {
 	int err = 0;
+	struct mbuf *mb = mbuf_alloc(512);
+	struct re_printf pfh = {
+		vprintf_handler,
+		mb
+	};
```

Comment:
I think it is better to have straightforward and readable code.

I dont think that you get much performance benefit with this approach.

In general, clean and maintainable code should be preferred over performance.

Reply:
@sreimers once wrote that printing to stdout may block sometimes.

I don't want to print to stdout while the mutex is locked. But I agree also with you and there might be a simpler solution with an mbuf. A commit will follow.